share/completion/git-restore

```diff
@@ -0,0 +1,48 @@
+# (C) 2024 unrealapex
+
+# Completion script for the "git-restore" command.
+
+function completion/git-restore {
+        WORDS=(git restore "${WORDS[2,-1]}")
+        command -f completion//reexecute
+}
+
+function completion/git::restore:arg {
+
+        OPTIONS=( #>#
+        "s --source=; restore the working tree files with the content from the given tree"
+        "p --patch; interactively select hunks in the difference between the restore source and the restore location"
+        "W --worktree; specify the restore location"
+        "S --staged; specify the restore location"
+        "q --quiet; quiet, supress feedback messages"
+        "--progress; display progress status"
+        "--no-progress; do not display progress status"
+        "--ours; use stage #2 (ours) when restoring files in the working tree from the index"
+        "--theirs; use stage #3 (theirs) when restoring files in the working tree from the index"
+        "m --merge; when restoring files on the working tree from the index, recreate the conflicted merge in the unmerged paths"
+        "--conflict=; the same as the --merge option, but changes the way the conflicting hunks are presented, overriding the merge.conflictStyle configuration variable"
+        "--ignore-unmerged; when restoring files on the working tree from the index, do not abort the operation if there are unmerted entries and neither --ours, --theirs, --merge, or --conflict is specified"
+        "--ignore-skip-worktree-bits; in sparse checkout mode, the default is to only update entries matched by <pathspec> and sparse patterns in \$GIT_DIR/info/sparse-checkout."
+        "--recurse-submodules; if <pathspec> names an active submodule and the restore location includes the working tree"
```

Comment:
These lines are longer than 80 characters. Would you like me to break them up?

Reply:
No, breaking lines would not work well with indentation...

Reply:
Is there any way you would like me to shorten the descriptions or are they find a they are?

Reply:
I used to rewrite descriptions myself when working on other completion scripts, but now we can ask ChatGPT to do the rewriting for us. Preferably, each description should be about 60 columns or shorter, although this is not a strict rule.

Reply:
ZSH completion provides very good descriptions that fit your criteria. I think using those would provide better descriptions than what ChatGPT could generate. I will update the completion descriptions.

Reply:
If you would like, I could update the descriptions of all of the Git completion scripts to use the more verbose descriptions.